### PR TITLE
[sdl2-image] Bump to 2.6.2.

### LINKED ISSUE #25284: [sdl2-image/mixer/net/ttf] Update to exported targets names.
****

## Description

Recently, a major CMake rework was done in the SDL2 subprojects, adding a consistent naming for the targets. Unfortunately, the naming chosen is inconsistent with the one used by vcpkg's ports.

The new naming consists in:
- The namespace `SDL2_XXX::` where `XXX` is the name of the subproject
- A static target `SDL2_XXX::SDL_XXX-static` linked to `SDL2::SDL2-static`
- A dynamic target `SDL2_XXX::SDL_XXX` linked to `SDL2::SDL2`

## Proposed solution

Adopt the naming chosen by the maintainers.

## Alternatives

To avoid breaking existing projects/ports, the targets currently in use by vcpkg could be kept as an alias of the new one with a deprecation message, similar to what is currently done in [SDL_ttf](https://github.com/libsdl-org/SDL_ttf/blob/fdb59157330bd0a8b0143d148a387823f2cc1c78/CMakeLists.txt#L401-L405).

## Additional context

Related PRs:
- https://github.com/libsdl-org/SDL_image/pull/238
- https://github.com/libsdl-org/SDL_mixer/pull/410
- https://github.com/libsdl-org/SDL_net/pull/48
- https://github.com/libsdl-org/SDL_ttf/pull/213

### PR DESCRIPTION
**Pull request by FtZPetruska**
Update SDL image from 2.0.5 to 2.6.2 bringing first party CMake support.

Additionally, moved to using `vcpkg_from_github` to enable using `--head` option and added a usage file.

Lastly, the fix for #21331 seems no longer needed, I was not able to reproduce the described issue on a Linux machine or a MacBook.

- #### What does your PR fix?
  Fixes #25284.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Unchanged.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

### CODE REVIEW

File: ports/sdl2-image/portfile.cmake [hunk 1]
```diff
@@ -1,43 +1,54 @@
...
-vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
-        libjpeg-turbo USE_JPEG
-        tiff USE_TIFF
-        libwebp USE_WEBP
+        libjpeg-turbo SDL2IMAGE_JPG
+        libwebp SDL2IMAGE_WEBP
+        tiff SDL2IMAGE_TIF
```
Comment (dg0yt) [inline ports/sdl2-image/portfile.cmake:17]: Please try to put the second keywords in a visual column.

File: ports/sdl2-image/portfile.cmake [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
 )
 
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
-vcpkg_cmake_config_fixup()
+if(EXISTS "${CURRENT_PACKAGES_DIR}/cmake")
+    vcpkg_cmake_config_fixup(PACKAGE_NAME SDL2_image CONFIG_PATH cmake)
+elseif(EXISTS "${CURRENT_PACKAGES_DIR}/SDL2_image.framework/Resources")
+    vcpkg_cmake_config_fixup(PACKAGE_NAME SDL2_image CONFIG_PATH SDL2_image.framework/Resources)
```
Comment (dg0yt) [inline ports/sdl2-image/portfile.cmake:39]: Is there no option to disable the framework stuff?

File: ports/sdl2pp/fix-dependencies.patch [hunk 3]
```diff
@@ -35,11 +35,10 @@ index cdfd2a6..dc08748 100644
  ENDIF(MINGW)
  
  IF(SDL2PP_WITH_IMAGE)
--	FIND_PACKAGE(SDL2_image REQUIRED)
+  FIND_PACKAGE(SDL2_image REQUIRED)
```
Comment (dg0yt) [inline ports/sdl2pp/fix-dependencies.patch:38]: Indentation?

Comment (dg0yt) [reply ports/sdl2pp/fix-dependencies.patch:38]: I see, it is context.

File: ports/sdl2-image/fix-pkgconfig.patch [hunk 4]
```diff
@@ -0,0 +1,64 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9abeefb..903e552 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -694,37 +694,34 @@ if(SDL2IMAGE_INSTALL)
+         COMPONENT devel
+     )
+ 
+-    if(SDL2IMAGE_BUILD_SHARED_LIBS)
```
[the remark this answers is not in the document]
Comment (FtZPetruska) [reply ports/sdl2-image/fix-pkgconfig.patch:9]: I stripped the patch to the strict minimum:
- Make the statement `if(1)` like you suggested.
- Keep the changes in `install` statement like the patches for sdl2-ttf/sdl2-mixer do.

File: ports/sdl2-image/portfile.cmake (hunk 2 above)
Comment (FtZPetruska) [reply ports/sdl2-image/portfile.cmake:39]: There is an [upstream patch](https://github.com/libsdl-org/SDL_image/commit/ede6056d844e7a1ecec3e688cb4068d1ad0aa884) that allows customising the CMake install directory, but it is not yet shipped in any release.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Comment by JavierMatosD** [comment]
Thanks!

**Pull request merged**